lib/assets/javascripts/cartodb/table/menu_modules/filters/histogram.js

```diff
@@ -255,8 +256,18 @@
           var upperLimit = this.model.get('upper_limit');
           if (value < lowerLimit) { value = lowerLimit; }
           if (value > upperLimit) { value = upperLimit; }
-          if (isMin && value > upper) { value = upper; }
-          if (!isMin && value < lower) { value = lower; }
+          if (isMin && value > upper) {
+            temp = upper;
+            upper = value;
+            value = temp;
+            this.model.set('upper', upper);
```

Comment:
This took me a moment to follow.  Not sure if 
```
            this.model.set('upper', value);
            value = upper;
```
is any better, but I think it achieves the same thing.  A comment explaining the purpose of these two bits might not go amiss.